### ISSUE TITLE
mypy does not pick up missing imports in installed stubs

### ISSUE THREAD

**Issue by Dr-Irv** [python/mypy#15169]
**Bug Report**
If you install a stubs package that has imports, and those imports are missing, then mypy doesn't report that those imports inside the installed stubs are missing

**To Reproduce**

- Create a clean python environment.
- Install `pandas-stubs` version 2.0.1.230501 from pypi
- Make sure that `numpy` is not installed
- Use this python code:
```python
import pandas as pd
```

Then no error is reported, even though `pandas-stubs` has an import for `numpy`

**Expected Behavior**

An error should be reported indicating that the stubs have a missing import.

**Actual Behavior**

No error is reported.

**Your Environment**

- Mypy version used:  1.2.0
- Mypy command-line flags: `--no-incremental`  Also tried `--follow-imports=error`
- Mypy configuration options from `mypy.ini` (and other config files):  None
- Python version used:  3.9.13

We could add `numpy` as a dependency in `pandas-stubs`, but I would have thought that `mypy` would have indicated an error when people are just installing stub files for type checking.  See https://github.com/pandas-dev/pandas-stubs/issues/672

**Comment by JelleZijlstra** [comment]
This is intentional, because for most users errors in stubs would not be actionable. If you're an end user who sees an error somewhere in `pandas-stubs`, you usually can't do anything about it other than update the package.

Possibly import errors should be treated differently, but I'm not sure that's worth the complexity.

**Comment by Dr-Irv** [comment]
> This is intentional, because for most users errors in stubs would not be actionable. If you're an end user who sees an error somewhere in `pandas-stubs`, you usually can't do anything about it other than update the package.
> 
> Possibly import errors should be treated differently, but I'm not sure that's worth the complexity.

The issue here is that the user didn't install a dependency that is imported by the stubs - although the dependency was created by the stubs.  The use case is people wanting to lint their code without installing full packages.

**Comment by AlexWaygood** [comment]
> The issue here is that the user didn't install a dependency that is imported by the stubs - although the dependency was created by the stubs. The use case is people wanting to lint their code without installing full packages.

Not sure I understand. If the stubs package explicitly lists numpy as a dependency, but numpy wasn't installed when the stubs package was installed, that sounds like a packaging bug.

**Comment by JelleZijlstra** [comment]
So you want people to be able to type-check pandas code without installing numpy, but you also want mypy to show an error if people are importing pandas without having numpy installed. That seems contradictory.

**Comment by AlexWaygood** [comment]
Everything is working as designed here, so this is a feature request rather than a bug report.

For the vast majority of users, it would be extremely frustrating if mypy started emitting errors while checking their code due to unrelated typing mistakes in libraries they'd installed in `site-packages`.

I suppose it's possible that mypy could start distinguishing between different kinds of typing errors in `site-packages` libraries ("the `site-packages` library is importing another library that doesn't exist" is a different kind of error to "the site packages library is returning `str` when the type annotation says it returns `int`"). But that could be difficult, error-prone, and would probably require a significant redesign. I'm not sure it's worth the effort.

I think it would probably be best for you to just bite the bullet and declare a dependency on `numpy`. Or add it as an optional dependency, and advertise that users won't get precise types unless they install `pandas-stubs` with the `numpy` extra.

**Comment by Dr-Irv** [comment]
> For the vast majority of users, it would be extremely frustrating if mypy started emitting errors while checking their code due to unrelated typing mistakes in libraries they'd installed in `site-packages`.

No, I'm suggesting that mypy emits errors while checking their code due to unrelated typing mistakes in *stubs* libraries (not libraries in general) they'd installed in `site-packages`.

**Comment by AlexWaygood** [comment]
> No, I'm suggesting that mypy emits errors while checking their code due to unrelated typing mistakes in _stubs_ libraries (not libraries in general) they'd installed in `site-packages`.

Why should there be a distinction between stubs packages mypy finds in `site-packages` and packages with inline types that it finds in `site-packages`. From mypy's perspective, both are PEP-561 Python packages, declared to be typed using a `py.typed` file.

**Comment by JelleZijlstra** [comment]
> > Why should there be a distinction between stubs packages mypy finds in `site-packages` and packages with inline types that it finds in `site-packages`. From mypy's perspective, both are PEP-561 Python packages, declared to be typed using a `py.typed` file. They work exactly the same way from a type checker's perspective. I'm not sure mypy treats the two any differently at all at the moment.
> 
> From the user point of view, I think there are differences between a pure stubs package (e.g. `pandas-stubs`), which has no `py.typed` file, versus a package like `numpy` which has a `py.typed` file and is mixing source code and `pyi` files. If you are using the former, it is purely for type checking. If you are using the latter, it is for type checking and runtime.
> 
> Again, the use case here is someone wanting to type-check their code by only installing stubs (e.g., the `--install-types` option of `mypy`). But if the installed stubs depend on a package providing typing within the source of the package, they have to then install that package.

Then pandas-stubs should depend on numpy. I still don't understand why you don't want to have that dependency, but you do want mypy to nudge users to install numpy if they use pandas-stubs. That's what dependencies are for. 

> since we have seen errors when installed that don't show up when testing locally.

That sounds like a bug. If you have a precise reproduction case, please report it.

**Comment by erictraut** [comment]
@AlexWaygood, PEP 561 [says the following](https://peps.python.org/pep-0561/#stub-only-packages) about stub-only packages (those that end in "-stubs"):

> Note that for stub-only packages adding a py.typed marker is not needed since the name *-stubs is enough to indicate it is a source of typing information.

**Comment by AlexWaygood** [comment]
> @AlexWaygood, PEP 561 [says the following](https://peps.python.org/pep-0561/#stub-only-packages) about stub-only packages (those that end in "-stubs"):
> 
> > Note that for stub-only packages adding a py.typed marker is not needed since the name *-stubs is enough to indicate it is a source of typing information.

Thanks, I'd misremembered that point.

**Comment by Dr-Irv** [comment]
> You are mistaken! If I install `pandas-stubs` into a fresh, empty virtual environment, there's definitely a `py.typed` file inside the `venv-dir/Lib/site-packages/pandas-stubs` directory! Mypy wouldn't know to look in the directory for type hints if there _weren't_ a `py.typed` file there.

Actually, you were correct.  Seems like we do supply a `py.typed` file for `pandas-stubs`, but as pointed out, it isn't required

**Comment by Dr-Irv** [comment]
> 
> > since we have seen errors when installed that don't show up when testing locally.
> 
> That sounds like a bug. If you have a precise reproduction case, please report it.

It was about 9 months ago when I observed this, so reproducing would be pretty difficult!

The reverse is definitely the case, where we see bugs in the local test that don't appear when installed.

**Comment by tmke8** [comment]
> * Create a clean python environment.
> 
>     * Install `pandas-stubs` version 2.0.1.230501 from pypi
> 
>     * Make sure that `numpy` is not installed
> 
>     * Use this python code:
> 
> 
> ```python
> import pandas as pd
> ```

Note that when I do this and use the flag `--no-silence-site-packages` then I do get errors about `numpy` (and also `jinja2` and `openpyxl`):

<details>
<summary>mypy output</summary>

```
$ mypy pandas_test.py --no-silence-site-packages
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexers.pyi:1: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/computation/ops.pyi:1: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/window/expanding.pyi:29: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/window/ewm.pyi:7: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/window/ewm.pyi:42: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/io/formats/style_render.pyi:13: error: Cannot find implementation or library stub for module named "jinja2"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/groupby/groupby.pyi:6: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/groupby/ops.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/datetimelike.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/tseries/holiday.pyi:11: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:15: error: Cannot find implementation or library stub for module named "odf.opendocument"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:16: error: Library stubs not installed for "openpyxl.workbook.workbook"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:16: note: Hint: "python3 -m pip install types-openpyxl"
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:16: note: (or run "mypy --install-types" to install all missing stub packages)
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:18: error: Cannot find implementation or library stub for module named "pyxlsb.workbook"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:18: error: Cannot find implementation or library stub for module named "pyxlsb"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/excel/_base.pyi:20: error: Cannot find implementation or library stub for module named "xlrd.book"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:15: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:186: error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:186: error: Overloaded function signatures 2 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:186: error: Overloaded function signatures 2 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:188: error: Overloaded function signature 3 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:190: error: Overloaded function signature 4 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:201: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:201: error: Overloaded function signatures 1 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:201: error: Overloaded function signatures 1 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:203: error: Overloaded function signatures 2 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:203: error: Overloaded function signatures 2 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:209: error: Overloaded function signature 5 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:209: error: Overloaded function signature 5 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timestamps.pyi:230: error: Return type "Tuple[int, int, int]" of "isocalendar" incompatible with return type "_IsoCalendarDate" in supertype "date"  [override]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/nattype.pyi:7: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/strings.pyi:14: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/strings.pyi:44: error: Overloaded function signatures 2 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/resample.pyi:13: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/resample.pyi:83: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/resample.pyi:107: error: Overloaded function signatures 4 and 5 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/generic.pyi:17: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/generic.pyi:21: error: Cannot find implementation or library stub for module named "sqlalchemy.engine"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/generic.pyi:21: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/formats/style.pyi:11: error: Cannot find implementation or library stub for module named "matplotlib.colors"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/formats/style.pyi:12: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/window/rolling.pyi:29: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/window/rolling.pyi:52: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/window/rolling.pyi:176: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/groupby/generic.pyi:15: error: Cannot find implementation or library stub for module named "matplotlib.axes"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/groupby/generic.pyi:59: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/core/groupby/generic.pyi:63: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:10: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 5 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 6 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 7 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 10 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 11 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:148: error: Overloaded function signatures 1 and 12 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 6 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 7 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 10 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 11 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:150: error: Overloaded function signatures 2 and 12 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:162: error: Overloaded function signature 8 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:162: error: Overloaded function signature 8 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:166: error: Overloaded function signature 9 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:166: error: Overloaded function signature 9 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:166: error: Overloaded function signature 9 will never be matched: signature 8's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:170: error: Overloaded function signature 10 will never be matched: signature 8's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:170: error: Overloaded function signature 10 will never be matched: signature 9's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:172: error: Overloaded function signature 11 will never be matched: signature 8's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:172: error: Overloaded function signature 11 will never be matched: signature 9's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:177: error: Overloaded function signature 12 will never be matched: signature 8's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:177: error: Overloaded function signature 12 will never be matched: signature 9's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:181: error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:181: error: Overloaded function signatures 2 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:181: error: Overloaded function signatures 2 and 6 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:181: error: Overloaded function signatures 2 and 7 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:183: error: Overloaded function signature 3 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:185: error: Overloaded function signature 4 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:185: error: Overloaded function signature 4 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:189: error: Overloaded function signature 5 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:189: error: Overloaded function signature 5 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:189: error: Overloaded function signature 5 will never be matched: signature 4's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:193: error: Overloaded function signature 6 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:193: error: Overloaded function signature 6 will never be matched: signature 4's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:193: error: Overloaded function signature 6 will never be matched: signature 5's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:195: error: Overloaded function signature 7 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:195: error: Overloaded function signature 7 will never be matched: signature 4's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:195: error: Overloaded function signature 7 will never be matched: signature 5's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:198: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:198: error: Overloaded function signatures 1 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:198: error: Overloaded function signatures 1 and 5 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:202: error: Overloaded function signature 3 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:206: error: Overloaded function signature 4 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:208: error: Overloaded function signature 5 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:212: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:212: error: Overloaded function signatures 1 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:212: error: Overloaded function signatures 1 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:212: error: Overloaded function signatures 1 and 5 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:212: error: Overloaded function signatures 1 and 6 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:212: error: Overloaded function signatures 1 and 9 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:214: error: Overloaded function signatures 2 and 3 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:214: error: Overloaded function signatures 2 and 4 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:214: error: Overloaded function signatures 2 and 5 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:214: error: Overloaded function signatures 2 and 6 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:214: error: Overloaded function signatures 2 and 9 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:224: error: Overloaded function signature 7 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:224: error: Overloaded function signature 7 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:228: error: Overloaded function signature 8 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:228: error: Overloaded function signature 8 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:228: error: Overloaded function signature 8 will never be matched: signature 7's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:232: error: Overloaded function signature 9 will never be matched: signature 7's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:232: error: Overloaded function signature 9 will never be matched: signature 8's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:244: error: Overloaded function signature 3 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:246: error: Overloaded function signature 4 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:248: error: Overloaded function signature 5 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:254: error: Signatures of "__rmul__" of "Timedelta" and "__mul__" of "Series[int]" are unsafely overlapping  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:254: error: Overloaded function signature 3 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:256: error: Signatures of "__rmul__" of "Timedelta" and "__mul__" of "Series[float]" are unsafely overlapping  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:256: error: Overloaded function signature 4 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:258: error: Overloaded function signature 5 will never be matched: signature 2's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:262: error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:262: error: Overloaded function signatures 1 and 5 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:262: error: Overloaded function signatures 1 and 6 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:262: error: Overloaded function signatures 1 and 7 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:262: error: Overloaded function signatures 1 and 8 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:262: error: Overloaded function signatures 1 and 9 overlap with incompatible return types  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:266: error: Overloaded function signature 3 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:270: error: Overloaded function signature 4 will never be matched: signature 1's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:270: error: Overloaded function signature 4 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:274: error: Overloaded function signature 5 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:274: error: Overloaded function signature 5 will never be matched: signature 4's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:278: error: Overloaded function signature 6 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:278: note: (Skipping most remaining errors due to unresolved imports or missing stubs; fix these first)
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:280: error: Overloaded function signature 7 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:282: error: Overloaded function signature 8 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/timedeltas.pyi:284: error: Overloaded function signature 9 will never be matched: signature 3's parameter type(s) are the same or broader  [misc]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/lib.pyi:7: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/period.pyi:7: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/interval.pyi:9: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/accessors.pyi:9: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/accessors.pyi:10: error: Cannot find implementation or library stub for module named "numpy.typing"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/range.pyi:4: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/multi.pyi:11: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/category.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/masked.pyi:1: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/base.pyi:15: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/groupby/grouper.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_typing.pyi:20: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/base.pyi:6: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/period.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/numpy_.pyi:1: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/numpy_.pyi:2: error: Cannot find implementation or library stub for module named "numpy.lib.mixins"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/interval.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/datetimes.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/categorical.pyi:11: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/base.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_misc.pyi:10: error: Cannot find implementation or library stub for module named "matplotlib.axes"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_misc.pyi:11: error: Cannot find implementation or library stub for module named "matplotlib.colors"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_misc.pyi:12: error: Cannot find implementation or library stub for module named "matplotlib.figure"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_misc.pyi:13: error: Cannot find implementation or library stub for module named "matplotlib.table"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_misc.pyi:14: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_core.pyi:14: error: Cannot find implementation or library stub for module named "matplotlib.axes"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_core.pyi:15: error: Cannot find implementation or library stub for module named "matplotlib.colors"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_core.pyi:16: error: Cannot find implementation or library stub for module named "matplotlib.lines"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_core.pyi:17: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/plotting/_core.pyi:22: error: Cannot find implementation or library stub for module named "scipy.stats.kde"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/_libs/tslibs/offsets.pyi:15: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/sql.pyi:15: error: Cannot find implementation or library stub for module named "sqlalchemy.engine"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/sql.pyi:15: error: Cannot find implementation or library stub for module named "sqlalchemy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/sql.pyi:16: error: Cannot find implementation or library stub for module named "sqlalchemy.sql.expression"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/io/sql.pyi:16: error: Cannot find implementation or library stub for module named "sqlalchemy.sql"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/series.pyi:23: error: Cannot find implementation or library stub for module named "core.api"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/series.pyi:29: error: Cannot find implementation or library stub for module named "matplotlib.axes"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/series.pyi:33: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/series.pyi:78: error: Cannot find implementation or library stub for module named "xarray"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/series.pyi:1956: note: Error code "misc" not covered by "type: ignore" comment
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexing.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/frame.pyi:20: error: Cannot find implementation or library stub for module named "matplotlib.axes"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/frame.pyi:21: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/frame.pyi:51: error: Cannot find implementation or library stub for module named "xarray"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/construction.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/algorithms.pyi:7: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/dtypes/missing.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/dtypes/dtypes.pyi:8: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/tools/numeric.pyi:6: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/tools/datetimes.pyi:11: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/timedeltas.pyi:11: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/timedeltas.pyi:58: note: Error code "misc" not covered by "type: ignore" comment
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/period.pyi:4: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/interval.pyi:12: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/indexes/datetimes.pyi:11: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/boolean.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/arrow/dtype.pyi:1: error: Cannot find implementation or library stub for module named "pyarrow"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/reshape/tile.pyi:7: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/reshape/pivot.pyi:13: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/reshape/melt.pyi:3: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/sparse/array.pyi:1: error: Cannot find implementation or library stub for module named "numpy"  [import]
.venv/lib/python3.11/site-packages/pandas-stubs/core/arrays/sparse/array.pyi:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
.venv/lib/python3.11/site-packages/pandas-stubs/core/util/hashing.pyi:1: error: Cannot find implementation or library stub for module named "numpy"  [import]
```

</details>

**Comment by Dr-Irv** [comment]
That option `--no-silence-site-packages` does solve the problem.   Given it exists, I will close this issue.
